presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/authentication/KerberosAuthentication.java

```diff
@@ -109,4 +111,13 @@ private static Configuration createConfiguration(String principal, String keytab
             }
         };
     }
+
+    private static String getServerPrincipal(String principal, String hostname)
+    {
+        String[] components = principal.split("[/@]");
+        if (components.length != 3 || !components[1].equals(HOSTNAME_PATTERN)) {
+            return principal;
+        }
+        return String.format("%s/%s@%s", components[0], hostname.toLowerCase(ENGLISH), components[2]);
```

Comment:
I am not sure why hadoop code was lowering things here. In Microsoft Kerberos I suppose it shouldn't matter. In MIT Kerberos this could lead to a bug.